test/test_functionlenses.jl

```diff
@@ -124,6 +174,11 @@ end
     o = @optic 1/(1 + exp(-_))
     @test o(2) ≈ 0.8807970779778823
     @test @inferred(set(2, o, 0.999)) ≈ 6.906754778648465
+
+    # setting inverse
+    f = @set inverse(sin) = sqrt
+    @test f(2) == sin(2)
```

Comment:
I find this super confusing. My expectation would be that `f(x) == x^2` but this seems to test `f(x) = sin(x)`?

Reply:
Ok I see. To make the test less confusing could you so something like `myasin(x) = asin(x)+2pi` and use that instead of sqrt?